std/regex/package.d

```diff
@@ -439,9 +439,7 @@ template ctRegexImpl(alias pattern, string flags=[])
         cast(immutable) r.withFactory(new CtfeFactory!(BacktrackingMatcher, Char, func));
     struct Wrapper
     {
-        // allow code that expects mutable Regex to still work
-        // we stay "logically const"
-        @trusted @property auto getRe() const { return cast() staticRe; }
+        @property ref getRe() const { return staticRe; }
```

Comment:
I recall some code that broke because of this change on Jenkins auto-tester.

Dunno if we should could overload both by ref and bu value

Reply:
If that's the problem would replacing `ref` with `auto ref` fix it?

Reply:
> I recall some code that broke because of this change on Jenkins auto-tester.

Yeah. Higgs fails:

```
parser/lexer.d(548): Error: none of the overloads of 'match' are callable using argument types (Wrapper), candidates are:
parser/lexer.d(335):        parser.lexer.StrStream.match(wstring str)
parser/lexer.d(351):        parser.lexer.StrStream.match(Regex!wchar re)
parser/lexer.d(664): Error: none of the overloads of 'match' are callable using argument types (Wrapper), candidates are:
parser/lexer.d(335):        parser.lexer.StrStream.match(wstring str)
parser/lexer.d(351):        parser.lexer.StrStream.match(Regex!wchar re)
parser/lexer.d(687): Error: none of the overloads of 'match' are callable using argument types (Wrapper), candidates are:
parser/lexer.d(335):        parser.lexer.StrStream.match(wstring str)
parser/lexer.d(351):        parser.lexer.StrStream.match(Regex!wchar re)
parser/lexer.d(699): Error: none of the overloads of 'match' are callable using argument types (Wrapper), candidates are:
parser/lexer.d(335):        parser.lexer.StrStream.match(wstring str)
parser/lexer.d(351):        parser.lexer.StrStream.match(Regex!wchar re)
makefile:77: recipe for target 'test' failed
```

https://ci.dlang.io/blue/organizations/jenkins/dlang-org%2Fphobos/detail/PR-5981/1/pipeline